staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go

```diff
@@ -388,7 +388,7 @@ func (ts *azureTokenSourceDeviceCode) Refresh(token *azureToken) (*azureToken, e
 	}
 
 	if err := spt.Refresh(); err != nil {
-		return nil, fmt.Errorf("refreshing token: %v", err)
+		return nil, err
```

Comment:
can you add a comment here saying the caller expects `TokenRefreshError` from `Refresh()` to trigger prompt

Reply:
Sure

Reply:
`fmt.Errorf("refreshing token: %w", err)` would still provide context and let IsTokenRefreshError detect the returned error, right?

Reply:
@liggitt thanks for mentioning this. `autorest.IsTokenRefreshError` is based on interface testing of the err object, from what I can see of both `fmt.Errorf` and `IsTokenRefreshError`, the fmt.Errorf would actually return a wrapError which supports `Unwrap()`, yet this is not available directly to `IsTokenRefreshError`, from golang perspective would it be preferable if we implement this way or to call `IsTokenRefreshError(err.Unwrap())` from the caller side?

Reply:
IsTokenRefreshError uses errors.As, which makes use of Unwrap, right?

Reply:
Added a simple test to prove that's true @liggitt.

```golang
func TestIsTokenRefreshError(t *testing.T) {
	refreshErr := fakeTokenRefreshError{message: "FakeError happened when refreshing"}
	err := fmt.Errorf("oops: %w", refreshErr)
	if !autorest.IsTokenRefreshError(err) {
		t.Fatalf("autorest.IsTokenRefreshError(err) fails!")
	}
}
```
let me check again on what's going on

Reply:
Ahh what's going in is the original code was using %v instead of %w (facepalm), thanks for the direction @liggitt, let me try the %w approach which preserves info.